packages/salesforcedx-vscode-core/src/commands/forceApexExecute.ts

```diff
@@ -4,99 +4,158 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-
-import {
-  Command,
-  SfdxCommandBuilder
-} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ExecuteAnonymousResponse } from '@salesforce/salesforcedx-apex/packages/apex/lib';
 import {
   CancelResponse,
   ContinueResponse,
   ParametersGatherer
 } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
-import * as fs from 'fs';
+// tslint:disable-next-line:no-var-requires
+const fs = require('fs').promises;
 import * as path from 'path';
 import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { handleApexLibraryDiagnostics } from '../diagnostics';
 import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { telemetryService } from '../telemetry';
 import { getRootWorkspacePath, hasRootWorkspace } from '../util';
 import {
+  ApexLibraryExecutor,
   SfdxCommandlet,
-  SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './util';
 
-class ForceApexExecuteExecutor extends SfdxCommandletExecutor<{}> {
-  public build(data: TempFile): Command {
-    return new SfdxCommandBuilder()
-      .withDescription(nls.localize('force_apex_execute_document_text'))
-      .withArg('force:apex:execute')
-      .withFlag('--apexcodefile', data.fileName)
-      .withLogName('force_apex_execute')
-      .build();
-  }
-}
-
-class CreateApexTempFile implements ParametersGatherer<{ fileName: string }> {
+export class CreateApexTempFile
+  implements ParametersGatherer<{ fileName: string }> {
   public async gather(): Promise<
     CancelResponse | ContinueResponse<{ fileName: string }>
   > {
     if (hasRootWorkspace()) {
-      const fileName = path.join(
-        getRootWorkspacePath(),
-        '.sfdx',
-        'tools',
-        'tempApex.input'
-      );
-      const editor = await vscode.window.activeTextEditor;
-
+      const editor = vscode.window.activeTextEditor;
       if (!editor) {
         return { type: 'CANCEL' };
       }
 
-      let writeFile;
       const document = editor.document;
-
-      if (editor.selection.isEmpty) {
-        writeFile = await writeFileAsync(fileName, document.getText());
-      } else {
-        writeFile = await writeFileAsync(
-          fileName,
-          document.getText(editor.selection)
+      let fileName = document.uri.fsPath;
```

Comment:
Why do we need to store this as a file first ? What if we update the library to accept a stream or a buffer ?

Reply:
Yeah we could definitely update the library to accept a buffer or stream as well. I think it would be worth it to address this in a separate PR, once we've added the functionality into the library, thoughts?

Reply:
The update in the library is simple, I think it would be better to do it correctly now than add debt.